src/services/VotesService.ts

```diff
@@ -0,0 +1,99 @@
+import HttpService, { Response, Response2 } from 'src/services/HttpService';
+import {
+  DeleteProjectVotesRequestPayload,
+  DeleteProjectVotesResponsePayload,
+  ProjectVotesPayload,
+  ProjectVotesResponsePayload,
+  UpsertProjectVotesRequestPayload,
+  UpsertProjectVotesResponsePayload,
+  VotingRecordsData,
+} from 'src/types/Votes';
+
+/**
+ * Accelerator "voting" related services
+ */
+
+// TODO: remove this constant and promises below once BE is ready
+const RETURN_MOCK_VOTING_DATA = true;
+
+const ENDPOINT_VOTES = '/api/v1/accelerator/projects/{projectId}/votes';
+
+const httpVoting = HttpService.root(ENDPOINT_VOTES);
+
+let mockGetProjectVotesResponseData: ProjectVotesPayload;
+
+const deleteProjectVotes = async (
+  projectId: number,
+  payload: DeleteProjectVotesRequestPayload
+): Promise<Response2<DeleteProjectVotesResponsePayload>> =>
+  RETURN_MOCK_VOTING_DATA
+    ? new Promise((resolve) => {
+        setTimeout(() => {
+          resolve({ data: { status: 'ok' }, requestSucceeded: true });
+        }, 300);
+      })
+    : httpVoting.delete2<DeleteProjectVotesResponsePayload>({
+        urlReplacements: { '{projectId}': `${projectId}` },
+        entity: payload,
+      });
+
+const getProjectVotes = async (projectId: number): Promise<Response2<ProjectVotesPayload>> =>
+  RETURN_MOCK_VOTING_DATA
+    ? new Promise((resolve) => {
+        setTimeout(() => {
+          resolve({ data: mockGetProjectVotesResponseData, requestSucceeded: true });
+        }, 300);
+      })
+    : httpVoting.get<ProjectVotesResponsePayload, VotingRecordsData>(
+        {
+          urlReplacements: { '{projectId}': `${projectId}` },
+        },
+        (response) => ({ votes: response?.votes })
+      );
+
+const setProjectVotes = async (projectId: number, payload: UpsertProjectVotesRequestPayload): Promise<Response> =>
+  RETURN_MOCK_VOTING_DATA
+    ? new Promise((resolve) => {
+        setTimeout(() => {
+          resolve({ requestSucceeded: true });
+        }, 300);
+      })
+    : httpVoting.put2<ProjectVotesResponsePayload>({
+        urlReplacements: { '{projectId}': `${projectId}` },
+        entity: payload,
+      });
+
+const VotesService = {
+  deleteProjectVotes,
+  getProjectVotes,
+  setProjectVotes,
+};
+
+export default VotesService;
+
+// TODO: remove mock data once BE is ready
+mockGetProjectVotesResponseData = {
+  projectId: 1,
```

Comment:
@tommylau523 can we also get the projectName with this payload?

Reply:
As of now, users with global roles cannot fetch the project name using project API.

Reply:
Thanks @tommylau523 , saw the PR update.